src/Tests/Microsoft.Diagnostics.Monitoring.UnitTestApp/Scenarios/StacksWorker.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+
+namespace Microsoft.Diagnostics.Monitoring.UnitTestApp.Scenarios
+{
+    internal class StacksWorker : IDisposable
+    {
+        private EventWaitHandle _eventWaitHandle = new ManualResetEvent(false);
+
+        public class StacksWorkerNested<T>
+        {
+            public void DoWork<U>(U test, WaitHandle handle)
+            {
+                Console.WriteLine(test);
+                handle.WaitOne();
+            }
+        }
+
+        public void Work()
+        {
+            StacksWorkerNested<int> nested = new StacksWorkerNested<int>();
+
+            nested.DoWork<long>(5, _eventWaitHandle);
+        }
+
+        public void Dispose()
+        {
+            _eventWaitHandle.Set();
+            _eventWaitHandle.Dispose();
```

Comment:
Is it okay to immediately dispose right after signaling the handle? Maybe you need another handle that the DoWork method will signal and the this Dipose method will wait on with a short timeout.